compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -84,12 +89,34 @@ OMR::Power::MemoryReference::MemoryReference(
    _staticRelocation(NULL),
    _unresolvedSnippet(NULL),
    _conditions(NULL),
-   _flag(0)
+   _flag(0),
+   _label(NULL)
    {
```

Comment:
Trying to use the symref to store the label doesn't work for certain cases where there is already a different symref applied to a MemoryReference which should refer to a label. Take, for instance, the case of the StartPC symbol. The symref applied to the MemoryReference in that case should be the StartPC symref, but in order to use a PC-relative addi to load that address, the StartPC _label_ (which is being introduced by #5380) must be used to calculate the offset.

Doing things this way is also consistent with how the X codegen already represents its PC-relative references [1] and avoids extra complexity when trying to determine if a MemoryReference is a PC-relative reference to a label, as all we have to do is check `_label != NULL` rather than having to actually examine the symbol.

[1] https://github.com/eclipse/omr/blob/b99d3036d8a20592a04fdcd77bca176e794e326d/compiler/x/codegen/OMRMemoryReference.hpp#L95